lib/grp.gi

```diff
@@ -1586,6 +1586,45 @@ function( G )
     return comp;
 end );
 
+#############################################################################
+##
+#M  Socle( <G> )  . . . . . . . . . . . . . . . . for finite nilpotent groups
+##
+InstallMethod( Socle, "for finite nilpotent groups",
+              [ IsGroup ],
+              RankFilter( IsGroup and IsFinite and IsNilpotentGroup ) -
+              RankFilter( IsGroup ),
+  function(G)
+    local H, prodH;
+
+    if not IsFinite(G) or not IsNilpotentGroup(G) then
+      TryNextMethod();
+    fi;
```

Comment:
You are right, it should also check first for ````CanComputeSize```` first. 

Do you want a tst file on Socle, as well? At least for testing the nilpotent case?